sotoki/sotoki.py

```diff
@@ -1367,6 +1435,18 @@ def run():
     print(
         "starting sotoki scraper...{}".format(f"using {TMPFS_DIR}" if TMPFS_DIR else "")
     )
+    if arguments["--optimization-cache"] is not None:
+        s3 = KiwixStorage(arguments["--optimization-cache"])
+        if not s3.check_credentials(
+            list_buckets=True, failsafe=True, write=True, read=True
+        ):
+            raise AuthenticationError(
```

Comment:
We don't need this external exception here. Just use `ValueError` as this is most likely a problem with the user-provided URL.
It would be nice to include details on failures, see https://github.com/openzim/youtube/blob/master/youtube2zim/scraper.py#L350

``` py 
    def s3_credentials_ok(self):
        logger.info("testing S3 Optimization Cache credentials")
        self.s3_storage = KiwixStorage(self.s3_url_with_credentials)
        if not self.s3_storage.check_credentials(
            list_buckets=True, bucket=True, write=True, read=True, failsafe=True
        ):
            logger.error("S3 cache connection error testing permissions.")
            logger.error(f"  Server: {self.s3_storage.url.netloc}")
            logger.error(f"  Bucket: {self.s3_storage.bucket_name}")
            logger.error(f"  Key ID: {self.s3_storage.params.get('keyid')}")
            logger.error(f"  Public IP: {get_public_ip()}")
            return False
        return True
```

Reply:
Implemented it